vms/proposervm/block_test.go

```diff
@@ -84,3 +86,265 @@ func TestPostForkCommonComponents_buildChild(t *testing.T) {
+	defer func() {
+		require.NoError(proVM.Shutdown(ctx))
+	}()
+
+	// Build a post fork block. It'll be the parent block in our test cases
+	parentTime := time.Now().Truncate(time.Second)
+	proVM.Set(parentTime)
+
+	coreParentBlk := &snowman.TestBlock{
+		TestDecidable: choices.TestDecidable{
+			IDV:     ids.GenerateTestID(),
+			StatusV: choices.Processing,
+		},
+		BytesV:  []byte{1},
+		ParentV: coreGenBlk.ID(),
+		HeightV: coreGenBlk.Height() + 1,
+	}
+	coreVM.BuildBlockF = func(context.Context) (snowman.Block, error) {
+		return coreParentBlk, nil
+	}
+	coreVM.GetBlockF = func(_ context.Context, blkID ids.ID) (snowman.Block, error) {
+		switch {
+		case blkID == coreParentBlk.ID():
+			return coreParentBlk, nil
+		case blkID == coreGenBlk.ID():
+			return coreGenBlk, nil
+		default:
+			return nil, errUnknownBlock
+		}
+	}
+	coreVM.ParseBlockF = func(_ context.Context, b []byte) (snowman.Block, error) { // needed when setting preference
+		switch {
+		case bytes.Equal(b, coreParentBlk.Bytes()):
+			return coreParentBlk, nil
+		case bytes.Equal(b, coreGenBlk.Bytes()):
+			return coreGenBlk, nil
+		default:
+			return nil, errUnknownBlock
+		}
+	}
+
+	parentBlk, err := proVM.BuildBlock(ctx)
+	require.NoError(err)
+	require.NoError(parentBlk.Verify(ctx))
+	require.NoError(parentBlk.Accept(ctx))
+
+	// Make sure preference is duly set
+	require.NoError(proVM.SetPreference(ctx, parentBlk.ID()))
+	require.Equal(proVM.preferred, parentBlk.ID())
+	_, err = proVM.getPostForkBlock(ctx, parentBlk.ID())
+	require.NoError(err)
+
+	// Mark node as non validator
+	valState.GetValidatorSetF = func(context.Context, uint64, ids.ID) (map[ids.NodeID]*validators.GetValidatorOutput, error) {
+		var (
+			aValidator = ids.GenerateTestNodeID()
+
+			// a validator with a weight large enough to fully fill the proposers list
+			weight = uint64(proposer.MaxBuildWindows * 2)
+		)
+		return map[ids.NodeID]*validators.GetValidatorOutput{
+			aValidator: {
+				NodeID: aValidator,
+				Weight: weight,
+			},
+		}, nil
+	}
+
+	coreChildBlk := &snowman.TestBlock{
+		TestDecidable: choices.TestDecidable{
+			IDV:     ids.GenerateTestID(),
+			StatusV: choices.Processing,
+		},
+		BytesV:  []byte{2},
+		ParentV: coreParentBlk.ID(),
+		HeightV: coreParentBlk.Height() + 1,
+	}
+	coreVM.BuildBlockF = func(context.Context) (snowman.Block, error) {
+		return coreChildBlk, nil
+	}
+
+	{
+		// Set local clock before MaxVerifyDelay from parent timestamp.
+		// Check that child block is not built.
+		localTime := parentBlk.Timestamp().Add(proposer.MaxVerifyDelay - time.Second)
+		proVM.Set(localTime)
+
+		_, err := proVM.BuildBlock(ctx)
+		require.ErrorIs(errProposerWindowNotStarted, err)
+	}
+
+	{
+		// Set local clock exactly MaxVerifyDelay from parent timestamp.
+		// Check that child block is not built.
+		localTime := parentBlk.Timestamp().Add(proposer.MaxVerifyDelay - time.Second)
```

Comment:
I think this should just be `proposer.MaxVerifyDelay`

Reply:
Copy paste error.